src/diff/index.js

```diff
@@ -291,6 +306,11 @@ function diffElementNodes(
 	}
 
 	if (dom == null) {
+		if (excessDomChildren != null) {
+			const index = excessDomChildren.indexOf(dom);
+			if (index !== -1) excessDomChildren[index] = null;
```

Comment:
Maybe shorten the statements as they are in here? https://github.com/preactjs/preact/pull/2259/files#diff-a04f8653407a342eac2cd2ea6a5c3ff6R127

Reply:
done